__tests__/pet.test.js

```diff
@@ -0,0 +1,187 @@
+        expect(myPet.age).toEqual(0);
+    })
+})
+
+describe('growUp',()=>{
+    it('increments the age by 1',()=>{
+        const myPet = new Pet('Fido');
+        myPet.growUp();
+        expect(myPet.age).toEqual(1);
+        expect(myPet.hunger).toEqual(5);
+        expect(myPet.fitness).toEqual(7);
+    })
+})
+
+
+
+describe('fitness',()=>{
+  it('increases fitness by walking by 4',()=>{
+    const myPet=new Pet('Fido');
+    myPet.fitness=4;
+    myPet.walk();
+    expect(myPet.fitness).toEqual(8);
+  })
+
+})
+
+describe('feed',()=>{
+  it('throws an error if my pet is dead',()=>{
+    const myPet = new Pet('Fido');
+    myPet.age=30;
+    expect(()=>myPet.feed()).toThrow('Your pet is no longer alive :(')
+  });
+
+    it('decrease hunger level by 3, minimum hunger is 0',()=>{
+        const myPet = new Pet('Fido');
+        myPet.hunger=9;
+        myPet.age=0;
+        myPet.feed();
+        expect(myPet.hunger).toEqual(6);
+    });
+ 
+});
+
+
+describe('checkup',()=>{
+  it('returns statement if fitness is below 3',()=>{
+    const myPet = new Pet('Fido');
+    myPet.fitness=2;
+    myPet.hunger=4;
+    myPet.checkUp();
+    expect(myPet.checkUp()).toBe("I need a walk")
+  })
+
+  it('returns statement if hunger is over 5 ',()=>{
+    const myPet = new Pet('Fido');
+    myPet.fitness=7;
+    myPet.hunger=7;
+    myPet.checkUp();
+    expect(myPet.checkUp()).toEqual("I am hungry")
+  })
+  it('returns statement if hunger is over 5 and fitness is below 3 ',()=>{
+    const myPet = new Pet('Fido');
+    myPet.fitness=2;
+    myPet.hunger=7;
+    myPet.checkUp();
+    expect(myPet.checkUp()).toEqual("I am hungry AND I need a walk")
+  })
+  it('returns statement if hunger is over 5 and fitness is below 3 ',()=>{
+    const myPet = new Pet('Fido');
+    myPet.fitness=5;
+    myPet.hunger=4;
+    myPet.checkUp();
+    expect(myPet.checkUp()).toEqual("I feel great!")
+  })
+})
+
+describe('isAlive',()=>{
+  it('if fitness is 0 or less, returns false',()=>{
+    const myPet=new Pet ('Fido');
+    myPet.fitness=0
+  expect(myPet.isAlive).toEqual(false)
+  })
+
+it('if hunger is 10 or more, returns false',()=>{
+  const myPet=new Pet('Fido');
+  myPet.hunger=10;
+expect(myPet.isAlive).toEqual(false);
+})
+
+
+it('if age is 30 or more, returns false',()=>{
+  const myPet=new Pet('Fido');
+  myPet.age=30;
+ expect(myPet.isAlive).toEqual(false);
+})
+
+it('age,fitness and hunger are all fine, return true',()=>{
+  const myPet=new Pet('Fido');
+  myPet.age=29;
+  myPet.fitness=1;
+  myPet.hunger=9;
+ expect(myPet.isAlive).toEqual(true);
+})
+
+})
+
+describe('parent pet',()=>{
+  it('parent pet instance of Pet',()=>{
+    const parent=new Pet ('Fiona');
+    expect(parent.name).toEqual('Fiona');
+  })
+
+  it('adds child name to children array',()=>{
+    const parent=new Pet('Fiona');
+    parent.haveBaby('Lisa');
+    const child=new Pet('Lisa');
+    parent.children=['Lisa']
+    expect(child.name).toEqual('Lisa')
```

Comment:
the only lines that are actually relevant in this test are:
```
const child = new Pet('Lisa');
expect(child.name).toEqual('Lisa');
```
which, if you look at it, is the exact same as the test in lines 10-13 and lines 125 - 128. None of the other code here is doing anything. I've already suggested how you can change your `haveBaby` function to actually work as intended. To test this new functionality, this test should be written like:
```
const parent = new Pet('Fiona');
parent.haveBaby('Lisa');

expect(parent.children[0]).toBeInstanceOf(Object);
expect(parent.children[0].name).toBe('Lisa');
```